Converters/CSi/Speckle.Converters.CSiShared/ToSpeckle/Helpers/CsiFramePropertiesExtractor.cs

```diff
@@ -0,0 +1,149 @@
+using Speckle.Converters.Common;
+using Speckle.Converters.CSiShared.Utils;
+
+namespace Speckle.Converters.CSiShared.ToSpeckle.Helpers;
+
+/// <summary>
+/// Extracts properties common to frame elements across CSi products (e.g., Etabs, Sap2000)
+/// using the FrameObj API calls.
+/// </summary>
+/// <remarks>
+/// Design Decisions:
+/// - Individual methods preferred over batched calls due to:
+///   * Independent API calls with no performance gain from batching (?)
+///   * Easier debugging and error tracing
+///   * Simpler maintenance as each method maps to one API concept
+/// Integration:
+/// - Part of the property extraction hierarchy
+/// - Used by <see cref="CsiGeneralPropertiesExtractor"/> for delegating frame property extraction
+/// </remarks>
+public sealed class CsiFramePropertiesExtractor
+{
+  private readonly IConverterSettingsStore<CsiConversionSettings> _settingsStore;
+
+  public CsiFramePropertiesExtractor(IConverterSettingsStore<CsiConversionSettings> settingsStore)
+  {
+    _settingsStore = settingsStore;
+  }
+
+  public void ExtractProperties(CsiFrameWrapper frame, Dictionary<string, object?> properties)
+  {
+    properties["applicationId"] = GetApplicationId(frame);
+
+    var geometry = DictionaryUtils.EnsureNestedDictionary(properties, "Geometry");
+    (geometry["startJointName"], geometry["endJointName"]) = GetEndPointNames(frame);
+
+    var assignments = DictionaryUtils.EnsureNestedDictionary(properties, "Assignments");
+    assignments["groups"] = new List<string>(GetGroupAssigns(frame));
+    assignments["materialOverwrite"] = GetMaterialOverwrite(frame);
+    assignments["localAxis"] = GetLocalAxes(frame);
+    assignments["propertyModifiers"] = GetModifiers(frame);
+    assignments["endReleases"] = GetReleases(frame);
+    assignments["sectionProperty"] = GetSectionName(frame);
+    assignments["path"] = GetPathType(frame);
+  }
+
+  private string GetApplicationId(CsiFrameWrapper frame)
+  {
```

Comment:
I've started using extension methods for speckle applicationIds: might make sense to do the same for your wrapper classes, where `wrapper.GetSpeckleApplicationId => SapModel.WrapperObj.GetGUID(wrapper.Name)`

Reply:
Added! :) However, with the following differences:

- Csi, unlike Autocad, doesn't work with native types (`Entity`, `DBObject` etc.)
- We have converter-specific wrapper types which are defined in the **Converters** project and part of that domain
- Therefore, the `GetSpeckleApplicationId` (`SpeckleApplicationIdExtensions`) added in Converters project